### DISCUSSION TITLE
Is there a way to get all the results in detail of a workflow through the API?

### DISCUSSION THREAD

**Discussion by ggalindo-networkingdev** [afourmy/eNMS#382] [General]
Greetings:
I’m having a problem when I try to run a workflow through the **eNMS API**. I used the` /rest/run_service `endpoint to run the workflow and it runs without trouble within eNMS, however, I can not find a way to get all the details and data of the workflow results. The only result I get, only shows the top-level results, for example, if the workflow ran successfully or failed.  The endpoint I used to receive the answer is: `/rest/result/<service_name>/<runtime>`
I also tried a different endpoint to test if I could get the workflow results using a query:` /rest/search`, However, that way I could not get them either.
**Is there a way to get all the results in detail of a workflow through the API?**
`</runtime></service_name>`

**Comment by afourmy** [comment]
`/rest/search` does just that if you provide the right payload.
e.g
```
{
    "type": "result",
    "columns": ["runtime", "parent_runtime", "result", "service_name", "workflow_name", "device_name"],
    "maximum_return_records": 100,
    "search_criteria": {
        "service_name": "NAME",
        "workflow_name": "NAME",
        "device_name": ".*",
        "device_name_filter": "regex",
        "device_name_invert": true,
        "parent_runtime": "2022-01-04 21:57:52.684233"
    },
    "order": [
        {"column": 0, "dir": "asc"}
    ]
}
```


by tweaking the values of `service_name`, `workflow_name`, and `device_name` you can get the exact results you're looking for
